src/app/lists/dataview.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+
+import Table from "@/components/Tables/Table";
+import supabase from "@/supabase";
+import React from "react";
+import { styled } from "styled-components";
+
+const TableDiv = styled.div`
+    margin: 20px;
+    padding: 20px;
+    background-color: ${(props) => props.theme.surfaceBackgroundColor};
+
+    border: solid 1px ${(props) => props.theme.surfaceForegroundColor};
+    border-radius: 1rem;
+`;
+
+const StyledTable = styled(Table)`
+    width: 100%;
+    background-color: transparent;
+`;
+
+type ListRow = { [headerKey: string]: string };
+
+const ListsDataView = async (): Promise<React.ReactElement> => {
+    // const rawData = (await supabase.from("lists").select("*")).data;
+    const rawData = [];
+    for (let i = 0; i < 100; i++) rawData.push({ item_name: `Item ${i}` });
+
+    console.log("fetched data");
+
+    const dataAndTooltips = rawData?.map((row) => {
+        const data: ListRow = {
+            item_name: row.item_name,
+        };
+        const tooltips: ListRow = {};
+
+        for (let i = 1; i <= 10; i++) {
+            const header = `${i}_quantity`;
+            // type cast required as the type system can't infer that the header is a key of row
```

Comment:
would this be fixed by changing L26 (init to empty list) to be
```tsx
const rawData: ListRow[] = [];
```
I really don't like `as` unless absolutely necessary, since we should be encoding types 'well enough' at all points. I think this is just TS going 'empty list could be a list of anything'.

Reply:
as has been removed